File: Backend/auth-service/utils/response.js
const { http } = require('http')

const RespondJSONSuccess = (res, statusCode, message, data) => {
    res.status(statusCode).json({
        status: http.STATUS_CODES[statusCode],
        message,
        data
    });
}

const RespondJSONError = (res, statusCode, message, errors = null) => {
    res.status(statusCode).json({
        status: http.STATUS_CODES[statusCode],
        message,
        errors
    });
}

module.exports = {
    RespondJSONSuccess,
    RespondJSONError
}

File: Backend/quote-service/utils/response.go
package utils

import (
	"net/http"

	"github.com/gin-gonic/gin"
)

func RespondJSON(c *gin.Context, status int, message string, data interface{}) {
	c.JSON(status, gin.H{
		"status":  http.StatusText(status),
		"message": message,
		"data":    data,
	})
}

func RespondError(c *gin.Context, status int, message string, err error) {
	c.JSON(status, gin.H{
		"status":  http.StatusText(status),
		"message": message,
		"errors":  err,
	})
}
